linux/hci/socket/socket.go

```diff
@@ -119,14 +119,21 @@ func open(fd, id int) (*Socket, error) {
 }
 
 func (s *Socket) Read(p []byte) (int, error) {
+	s.rmu.Lock()
+	n, err := unix.Read(s.fd, p)
+	s.rmu.Unlock()
+	// Close always sends a dummy command to wake up Read
```

Comment:
This comment is a little unclear. Maybe `Read` needs to be removed?

Reply:
would changing "wake up Read" to "wake this function up" help?

Reply:
I guess? Do I understand this?
When the socket is closed, a dummy command is always sent to wake up the socket read function.